src/events.ts

```diff
@@ -27,27 +25,32 @@ export type ListenTo<EventMap> = Pick<
   "addListener" | "removeListener" | "on" | "off"
 >
 
-
 export type FileSystem = {
   "fileSystem:local-change": { dataRoot: CID; path: DistinctivePath<Partitioned<Partition>> }
   "fileSystem:publish": { dataRoot: CID }
 }
 
-
-export type Session<S> = {
-  "session:create": { session: S }
-  "session:destroy": { username: string }
+export type CapabilityConsumer = {
+  "capabilities:challenge": any // TODO
 }
 
+export type CapabilityProvider = {
+  "capabilities:approved": void
+  "capabilities:challenge": any // TODO
+  "capabilities:dismissed": void
+  "capabilities:query": { capabilities: Record<string, any> } // TODO
+}
```

Comment:
Will these be renamed to `authority`? Also, will consumer be renamed to requestor? That would align with the terms used in the new AWAKE spec.

Understood this code is WIP, so no need to change anything on this PR. 🙂

Reply:
Ah yes, forgot to rename these. I'll change it to requestor 👍